src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -19,29 +19,40 @@ public class DeleteCommand extends Command {
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
+            + ": Deletes the person identified by the NRIC number used in the displayed person list.\n"
+            + "Parameters: NRIC (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
-    private final Index targetIndex;
+    private final Nric targetNric;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(Nric targetNric) {
+        this.targetNric = targetNric;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        boolean found = false;
+        Person personToDelete = null;
+
+        for (Person person : lastShownList) {
+            if (person.getNric().equals(targetNric)) {
+                found = true;
+                personToDelete = person;
+                break;
+            }
+        }
+
```

Comment:
Hi @cliftonchee, you can abstract this away with the `hasPersonWithNric` method in `AddressBook.java`